examples/async-futures-select/src/worker.rs

```diff
@@ -54,7 +54,9 @@ impl Worker {
                     !pending_tasks.is_empty()
                 }
             };
-            if try_dispatch {
+
+            // Check that all TaskRequest and CompFinished at the current time are processed.
+            if try_dispatch && self.ctx.yield_now().now_or_never().is_some() {
```

Comment:
```suggestion
            // Check that all events at the current time are received.
            // Otherwise, we can miss a CompFinished event during task dispatching.
            let all_received = self.ctx.yield_now().now_or_never().is_some();
            if try_dispatch && all_received {
```